src/sig_stateful/sig_stfl.h

```diff
@@ -0,0 +1,309 @@
+#define OQS_SIG_STFL_alg_xmssmt_sha256_2_h40 "XMSSMT-SHA2_40/2_256"
+/* Algorithm identifier for XMSSMT-SHA2_40/4_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_4_h40 "XMSSMT-SHA2_40/4_256"
+/* Algorithm identifier for XMSSMT-SHA2_40/8_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_8_h40 "XMSSMT-SHA2_40/8_256"
+/* Algorithm identifier for XMSSMT-SHA2_60/3_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_3_h60 "XMSSMT-SHA2_60/3_256"
+/* Algorithm identifier for XMSSMT-SHA2_60/6_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_6_h60 "XMSSMT-SHA2_60/6_256"
+/* Algorithm identifier for XMSSMT-SHA2_60/12_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_12_h60 "XMSSMT-SHA2_60/12_256"
+
+#define OQS_SIG_STFL_algs_length 20
+
+/**
+ * Returns identifiers for available signature schemes in liboqs.  Used with OQS_SIG_STFL_new.
+ *
+ * Note that algorithm identifiers are present in this list even when the algorithm is disabled
+ * at compile time.
+ *
+ * @param[in] i Index of the algorithm identifier to return, 0 <= i < OQS_SIG_algs_length
+ * @return Algorithm identifier as a string, or NULL.
+ */
+OQS_API const char *OQS_SIG_STFL_alg_identifier(size_t i);
+
+/**
+ * Returns the number of signature mechanisms in liboqs.  They can be enumerated with
+ * OQS_SIG_STFL_alg_identifier.
+ *
+ * Note that some mechanisms may be disabled at compile time.
+ *
+ * @return The number of signature mechanisms.
+ */
+OQS_API int OQS_SIG_STFL_alg_count(void);
+
+/**
+ * Indicates whether the specified algorithm was enabled at compile-time or not.
+ *
+ * @param[in] method_name Name of the desired algorithm; one of the names in `OQS_SIG_STFL_algs`.
+ * @return 1 if enabled, 0 if disabled or not found
+ */
+OQS_API int OQS_SIG_STFL_alg_is_enabled(const char *method_name);
+
+
+/**
+ * @brief OQS_SECRET_KEY object for stateful signature schemes
+ */
+typedef struct OQS_SECRET_KEY OQS_SECRET_KEY;
+
+typedef struct OQS_SECRET_KEY {
+
+	/* The (maximum) length, in bytes, of secret keys for this signature scheme. */
+	size_t length_secret_key;
+
+	/* The secret key stored in memory as an array of bytes*/
+	uint8_t *secret_key;
+
+	/* User defined data that may be used for the SAFETY functions */
+	void *data;
+
+	/* Function that returns the total number of signatures for the secret key */
+	unsigned long long (*sigs_total)(const OQS_SECRET_KEY *secret_key);
+
+	/* Function that returns the number of signatures left for the secret key */
+	unsigned long long (*sigs_left)(const OQS_SECRET_KEY *secret_key);
+
+	/**
+	 * SAFETY FUNCTIONS
+	 * -> lock_key : "Locks" the secret key, preventing any attributes from being changed
+	 * -> save_secret_key : Saves the byte array of the secret key to a file
+	 * -> release_key : "Unlocks" the secret key, allowing attributes to be changed
+	 */
+	OQS_STATUS (*lock_key)(OQS_SECRET_KEY *sk);
+	OQS_STATUS (*save_secret_key)(const OQS_SECRET_KEY *sk);
+	OQS_STATUS (*release_key)(OQS_SECRET_KEY *sk);
```

Comment:
This will probably fail the Doxygen requirements -- there would need to be documentation for each method.

Reply:
Ok I will fix it